File: src/main/intellibitz/sted/event/MessageEvent.java
package intellibitz.sted.event;

import java.util.EventObject;

/**
 * holds the source Thread information
 */
public class MessageEvent
        extends EventObject
        implements Cloneable {
    private IMessageEventSource messageEventSource;
    private String message;

    /**
     * @param src the source for this event
     */
    public MessageEvent(IMessageEventSource src) {
        super(src);
        messageEventSource = src;
    }

    /**
     * @return IStatusEventSource the Source which generated this Event
     */
    public IMessageEventSource getEventSource() {
        return messageEventSource;
    }

    public String getMessage() {
        return message;
    }

    public void setMessage(String message) {
        this.message = message;
    }

    public Object clone()
            throws CloneNotSupportedException {
        return super.clone();
    }
}

File: src/main/intellibitz/sted/event/MessageEvent.kt
package sted.event

import java.util.*

class MessageEvent(val eventSource: IMessageEventSource) : EventObject(eventSource), Cloneable {
    var message: String? = null

    @Throws(CloneNotSupportedException::class)
    public override fun clone(): Any {
        return super.clone()
    }
}